persistence/soft-index/src/test/java/org/infinispan/persistence/sifs/SoftIndexFileStoreTest.java

```diff
@@ -106,6 +113,55 @@ public void testStopStartAndMultipleWrites() {
       assertEquals("v2", store.load("k1").getValue());
    }
 
+   // test for ISPN-5743
+   public void testStopStartWithRemoves() {
```

Comment:
It seems this test (or maybe another from this class) is causing the suite to hang, please take a look at the stack trace https://gist.github.com/gustavonalle/72d470eb075ce6e0fbd9
I was able to reproduce quickly by running  ```mvn clean test``` in a loop.
Tested on master as well and it is not affected, ```SoftIndexFileStoreTest``` always passes